db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/JdbcTaskRepository.java

```diff
@@ -135,9 +135,9 @@ public void getScheduledExecutions(ScheduledExecutionsFilter filter, String task
     @Override
     public List<Execution> getDue(Instant now, int limit) {
         final UnresolvedFilter unresolvedFilter = new UnresolvedFilter(taskResolver.getUnresolved());
-
+        final String explicitLimit = jdbcCustomization.supportsExplicitQueryLimitPart() ? jdbcCustomization.getQueryLimitPart(limit) : "";
         return jdbcRunner.query(
-                "select * from " + tableName + " where picked = ? and execution_time <= ? " + unresolvedFilter.andCondition() + " order by execution_time asc",
+                "select * from " + tableName + " where picked = ? and execution_time <= ? " + unresolvedFilter.andCondition() + " order by execution_time asc" + explicitLimit,
```

Comment:
I would be tempted to drop `setMaxRows(...)` when explicit limit in sql is used

Reply:
Ok, I've removed `setMaxRows` when there is explicit limit